src/libraries/Microsoft.PowerFx.Interpreter/Environment/OptionSet.cs

```diff
@@ -20,16 +21,15 @@ namespace Microsoft.PowerFx
 {
     public class OptionSet : IExternalOptionSet
     {
-        private readonly SingleSourceDisplayNameProvider _displayNameProvider;
+        private readonly DisplayNameProvider _displayNameProvider;
         private readonly DType _type;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="OptionSet"/> class.
         /// </summary>
         /// <param name="name">The name of the option set. Will be available as a global name in Power Fx expressions.</param>
         /// <param name="options">The members of the option set. Enumerable of pairs of logical name to display name.
-        /// Consider using <see cref="DisplayNameUtility.MakeUnique(IEnumerable{KeyValuePair{string, string}})"/> 
-        /// to ensure that display and logical names for options are unique.
+        /// NameCollisionException is thrown if display and logical names for options are not unique.
         /// </param>
         public OptionSet(string name, IEnumerable<KeyValuePair<DName, DName>> options)
```

Comment:
Can this ctor just chain to the new one?

: this(name, new SingleSourceDisplayNamePRovider(options)) 

And then the body is empty - which means more code sharing.

Reply:
absolutely  :)